cmd/publicize/main.go

```diff
@@ -0,0 +1,174 @@
+		}
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+// bond holds the information of which repos are bonded to each other.
+type Bond struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+func (b *Bond) validate() error {
+	for _, value := range []string{b.From, b.To} {
+		if len(strings.Split(value, "/")) != 2 {
+			return fmt.Errorf("%s should be in org/repo format", value)
+		}
+	}
+	return nil
+}
+
+type options struct {
+	port int
+
+	configPath string
+	bonds      map[string]string
+	dryRun     bool
+	confirm    bool
+	github     prowflagutil.GitHubOptions
+
+	gitName           string
+	gitEmail          string
+	webhookSecretFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.IntVar(&o.port, "port", 8888, "Port to listen on.")
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to publicize configuration.")
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+	fs.BoolVar(&o.confirm, "confirm", false, "It true, perform an actual non-dry push to destination repository.")
+
+	fs.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email.")
+	fs.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name.")
+
+	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		group.AddFlags(fs)
+	}
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func (o *options) Validate() error {
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		if err := group.Validate(o.dryRun); err != nil {
+			return err
+		}
+	}
+
+	if o.gitName == "" {
+		return fmt.Errorf("git-name is not specified.")
+	}
+
+	if o.gitEmail == "" {
+		return fmt.Errorf("git-email is not specified.")
+	}
+
+	bytes, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return fmt.Errorf("Couldn't read publicize configuration file: %v", o.configPath)
+	}
+
+	var c Config
+	if err := yaml.Unmarshal(bytes, &c); err != nil {
+		return errors.New("Couldn't unmarshal publicize configuration")
+	}
+
+	if err := c.validate(); err != nil {
+		return err
+	}
+
+	o.bonds = make(map[string]string)
+	for _, bond := range c.Bonds {
+		o.bonds[bond.From] = bond.To
+	}
+
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	log := logrus.StandardLogger().WithField("plugin", "publicize")
+
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.github.TokenPath, o.webhookSecretFile}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	webhookTokenGenerator := secretAgent.GetTokenGenerator(o.webhookSecretFile)
```

Comment:
How much of this is shared with other plugins? Can we factor it out into a library in `test-infra`?

Reply:
It seems that all of the plugins are using that. I will open a follow-up to share this.